### PR TITLE
Feature/hw 66538 v4 stakeholder upload multipart

### PR DESCRIPTION
**Pull request by ramahalingam**

### CODE REVIEW

File: src/main/java/com/hyperwallet/clientsdk/model/VerificationDocument.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+import javax.xml.bind.annotation.XmlAccessType;
+import javax.xml.bind.annotation.XmlAccessorType;
+import javax.xml.bind.annotation.XmlRootElement;
+
+@JsonInclude(Include.NON_NULL)
+@XmlRootElement
+@XmlAccessorType(XmlAccessType.FIELD)
+public class VerificationDocument {
+
```
Comment (arao6) [inline src/main/java/com/hyperwallet/clientsdk/model/VerificationDocument.java:14]: Pls rename this class to HyperwalletDocumentVerification

File: src/test/java/com/hyperwallet/clientsdk/HyperwalletTest.java [hunk 2]
```diff
@@ -7042,5 +7041,65 @@ public void testUpdateVenmoAccount_successful() throws Exception {
...
+        } catch (HyperwalletException e) {
+            assertThat(e.getErrorCode(), is(nullValue()));
+            assertThat(e.getResponse(), is(nullValue()));
+            assertThat(e.getErrorMessage(), is(equalTo("User token may not be present")));
+            assertThat(e.getMessage(), is(equalTo("User token may not be present")));
+            assertThat(e.getHyperwalletErrors(), is(nullValue()));
+            assertThat(e.getRelatedResources(), is(nullValue()));
+        }
+    }
+
```
Comment (arao6) [inline src/test/java/com/hyperwallet/clientsdk/HyperwalletTest.java:7061]: Extra line can be remvoed

File: src/main/java/com/hyperwallet/clientsdk/model/VerificationDocument.java (hunk 1 above)
Comment (ramahalingam) [reply src/main/java/com/hyperwallet/clientsdk/model/VerificationDocument.java:14]: Done.

File: src/test/java/com/hyperwallet/clientsdk/HyperwalletTest.java (hunk 2 above)
Comment (ramahalingam) [reply src/test/java/com/hyperwallet/clientsdk/HyperwalletTest.java:7061]: Done

**Review by Rrathinasabapath — CHANGES_REQUESTED** [review]
Pls make necessary changes

File: src/main/java/com/hyperwallet/clientsdk/model/HyperwalletBusinessStakeholder.java [hunk 3]
```diff
@@ -641,4 +643,25 @@ public HyperwalletBusinessStakeholder clearCountry() {
...
+
+    public HyperwalletBusinessStakeholder documents(List<VerificationDocument> documents) {
+        addField("documents", documents);
+        this.documents = documents;
+        return this;
+    }
+
+    public HyperwalletBusinessStakeholder clearDocuments() {
+        clearField("documents");
+        documents = null;
```
[the remark this answers is not in the document]
Comment (ramahalingam) [reply src/main/java/com/hyperwallet/clientsdk/model/HyperwalletBusinessStakeholder.java:664]: Made change as suggested.

File: src/test/java/com/hyperwallet/clientsdk/HyperwalletTest.java [hunk 4]
```diff
@@ -7042,5 +7041,65 @@ public void testUpdateVenmoAccount_successful() throws Exception {
...
+
+    @Test
+    public void testUploadDocumentBusinessStakeholder_noBusinessStakeholderToken() {
+        Hyperwallet client = new Hyperwallet("test-username", "test-password");
+        String userToken = "business-token";
```
[the remark this answers is not in the document]
Comment (ramahalingam) [reply src/test/java/com/hyperwallet/clientsdk/HyperwalletTest.java:7066]: Changed to user-token

File: src/test/java/com/hyperwallet/clientsdk/HyperwalletTest.java [hunk 5]
```diff
@@ -7042,5 +7041,65 @@ public void testUpdateVenmoAccount_successful() throws Exception {
...
+        try {
+            HyperwalletBusinessStakeholder hyperwalletBusinessStakeholderResponse =
+                    client.uploadDocumentBusinessStakeholder(userToken, null, new FormDataMultiPart());
+            fail("Expect HyperwalletException");
+        } catch (HyperwalletException e) {
+            assertThat(e.getErrorCode(), is(nullValue()));
+            assertThat(e.getResponse(), is(nullValue()));
+            assertThat(e.getErrorMessage(), is(equalTo("BusinessStakeholderToken is required")));
+            assertThat(e.getMessage(), is(equalTo("BusinessStakeholderToken is required")));
```
[the remark this answers is not in the document]
Comment (ramahalingam) [reply src/test/java/com/hyperwallet/clientsdk/HyperwalletTest.java:7075]: Changed to BusinessStakeholderToken may not be required

File: src/test/java/com/hyperwallet/clientsdk/HyperwalletTest.java [hunk 6]
```diff
@@ -7042,5 +7041,65 @@ public void testUpdateVenmoAccount_successful() throws Exception {
...
+        verificationDocumentList.add(verificationDocument);
+        hyperwalletBusinessStakeholder.setDocuments(verificationDocumentList);
+
+        HyperwalletApiClient mockApiClient = createAndInjectHyperwalletApiClientMock(client);
+        Mockito.when(mockApiClient.put(Mockito.anyString(), Mockito.any(FormDataMultiPart.class), Mockito.any(Class.class)))
+                .thenReturn(hyperwalletBusinessStakeholder);
+
+        HyperwalletBusinessStakeholder hyperwalletBusinessStakeholderResponse =
+                client.uploadDocumentBusinessStakeholder(userToken, businessStakeholderToken, new FormDataMultiPart());
+        assertTrue(hyperwalletBusinessStakeholderResponse.getDocuments().get(0).getCategory().equals("IDENTIFICATION"));
```
[the remark this answers is not in the document]
Comment (ramahalingam) [reply src/test/java/com/hyperwallet/clientsdk/HyperwalletTest.java:7100]: Changed as suggested.

File: src/main/java/com/hyperwallet/clientsdk/util/HyperwalletApiClient.java [hunk 7]
```diff
@@ -62,6 +76,25 @@ public HyperwalletApiClient(final String username, final String password, final
         }
     }
 
+    public WebResource getWebResource(final String url) {
```
[the remark this answers is not in the document]
Comment (ramahalingam) [reply src/main/java/com/hyperwallet/clientsdk/util/HyperwalletApiClient.java:79]: Changed the method to be private.

**Review by Rrathinasabapath — APPROVED** [review]
Changes approved

**Review by vkorotchenko — APPROVED** [review]

**Pull request merged**